subprojects/snapshots/src/test/groovy/org/gradle/internal/snapshot/AbstractSnapshotWithChildrenTest.groovy

```diff
@@ -0,0 +1,208 @@
+        ("a".."z").toList(),
+    ]
+
+    /**
+     * The queried/updated path has no common prefix with any of the initial children of the node under test.
+     *
+     * E.g.
+     *   path: 'name0/some
+     *   children: ['name/some', 'other']
+     * or
+     *   path: 'completelyDifferent'
+     *   children: ['name/some', 'other']
+     */
+    static final List<VirtualFileSystemTestSpec> NO_COMMON_PREFIX = INITIAL_CHILD_CONSTELLATIONS.collectMany { childPaths ->
+        childPaths.collect { childPath ->
+            def firstSlash = childPath.indexOf('/')
+            String newChildPath = firstSlash > -1
+                ? "${childPath.substring(0, firstSlash)}0${childPath.substring(firstSlash)}"
+                : "${childPath}0"
+            new VirtualFileSystemTestSpec(childPaths, newChildPath, 0, null)
+        } + new VirtualFileSystemTestSpec(childPaths, 'completelyDifferent', 0, null)
+    } + new VirtualFileSystemTestSpec([], 'different', 0, null)
+
+    /**
+     * The queried/updated path has a true common prefix with one of the initial children of the node under test.
+     *
+     * E.g.
+     *   path: 'name/different'
+     *   children: ['name/some', 'other']
+     */
+    static final List<VirtualFileSystemTestSpec> COMMON_PREFIX = INITIAL_CHILD_CONSTELLATIONS.collectMany { childPaths ->
+        childPaths.findAll { it.contains('/') } collectMany { childPath ->
+            parentPaths(childPath).collect {
+                new VirtualFileSystemTestSpec(childPaths, "${it}/different", 0, childPath)
+            }
+        }
+    }
+
+    /**
+     * The queried/updated path is a prefix of one of the initial children of the node under test.
+     *
+     * E.g.
+     *   path: 'name'
+     *   children: ['name/some', 'other']
+     */
+    static final List<VirtualFileSystemTestSpec> IS_PREFIX_OF_CHILD = INITIAL_CHILD_CONSTELLATIONS.collectMany { childPaths ->
+        childPaths.findAll { it.contains('/') } collectMany { childPath ->
+            parentPaths(childPath).collect { parentPath ->
+                new VirtualFileSystemTestSpec(childPaths, parentPath, 0, findPathWithParent(childPaths, parentPath))
+            }
+        }
+    }
+
+    /**
+     * The queried/updated path is one of the initial children of the node under test.
+     *
+     * E.g.
+     *   path: 'name/some'
+     *   children: ['name/some', 'other']
+     */
+    static final List<VirtualFileSystemTestSpec> SAME_PATH = INITIAL_CHILD_CONSTELLATIONS.collectMany { childPaths ->
+        childPaths.collect {
+            new VirtualFileSystemTestSpec(childPaths, it, 0, it)
+        }
+    }
+
+    /**
+     * One of the initial children of the node under test is a prefix of the queried/updated path.
+     *
+     * E.g.
+     *   path: 'name/some/descendant'
+     *   children: ['name/some', 'other']
+     */
+    static final List<VirtualFileSystemTestSpec> CHILD_IS_PREFIX = INITIAL_CHILD_CONSTELLATIONS.collectMany { childPaths ->
```

Comment:
This is modifying a parent of a descendant.

Reply:
No. The selected path is a descendant of an existing child.